log/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use lv2_core::feature::*;
+//use std::ffi::CString;
+use std::fmt;
+use std::os::raw::*; //get all common c_type
+use urid::*;
+
+pub struct EntryClass;
+unsafe impl UriBound for EntryClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Entry;
+}
+
+pub struct ErrorClass;
+unsafe impl UriBound for ErrorClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Error;
+}
+
+pub struct NoteClass;
+unsafe impl UriBound for NoteClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Note;
+}
+
+pub struct TraceClass;
+unsafe impl UriBound for TraceClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Trace;
+}
+
+pub struct WarningClass;
+unsafe impl UriBound for WarningClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Warning;
+}
+
+/// Marker for URID representing the nature of a log message
+// Note : it's may be better to have a URID trait to define a common interface
+pub unsafe trait EntryType {
+    fn get(self) -> u32;
+}
+
+unsafe impl EntryType for URID<ErrorClass> {
+    fn get(self) -> u32 {
+        URID::<ErrorClass>::get(self)
+    }
+}
+unsafe impl EntryType for URID<NoteClass> {
+    fn get(self) -> u32 {
+        URID::<NoteClass>::get(self)
+    }
+}
+unsafe impl EntryType for URID<TraceClass> {
+    fn get(self) -> u32 {
+        URID::<TraceClass>::get(self)
+    }
+}
+unsafe impl EntryType for URID<WarningClass> {
+    fn get(self) -> u32 {
+        URID::<WarningClass>::get(self)
+    }
+}
+
+/// The Log feature
+#[repr(transparent)]
+pub struct Log<'a> {
+    internal: &'a lv2_sys::LV2_Log_Log,
+}
+
+unsafe impl<'a> UriBound for Log<'a> {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__log;
+}
+
+unsafe impl<'a> Feature for Log<'a> {
+    // Note: this feature can be used in any threading class and doesn't seems to have thready
+    // unsafty
+    unsafe fn from_feature_ptr(feature: *const c_void, _class: ThreadingClass) -> Option<Self> {
+        (feature as *const lv2_sys::LV2_Log_Log)
+            .as_ref()
+            .map(|internal| Self { internal })
+    }
+}
+
+impl<'a> Log<'a> {
+    pub fn print(&self, entry_type: impl EntryType, message: &[u8]) -> Result<(), ()> {
```

Comment:
Using a `&[u8]` seems unsafe, as it does not guarantee that it is null-terminated. Using `&CStr` here would probably be better. :slightly_smiling_face: 

Also, since this method is quite limited and not the full implementation of the underlying print, it should probably be named `print_cstr`, that way we can add a full `print` method later without it being a breaking change.

Reply:
i changed  this, now i create a String from the &str and add the missing "\0"